### PR TITLE
All: delete outdated deprecations

### PR DESCRIPTION
**Pull request by erik-krogh**
Similar to last time: https://github.com/github/codeql/pull/16330

A little more than last time, so also a slightly larger change-note (now marked as breaking).  

The swift CI failure is unrelated.

### CODE REVIEW

**Comment by jketema** [comment]
The deprecated C++ dataflow library should not be deleted, we still have a dependency on it.

**Review by jketema — DISMISSED** [review]
👍  C/C++ and Swift.

**Review by owen-mc — DISMISSED** [review]
Go 👍🏻

**Comment by dbartol** [comment]
This will bump the major version of all affected library packs. Any customer pack that depends on, say, `^1.1.0` of the library pack will stay stuck at the latest `1.x.y` version until the customer updates their dependency declaration. 

There's nothing wrong with this per se. Just be aware of the customer impact when you make these sorts of changes. Batching these up into a single breaking change is good, and you seem to have done that here.

**Review by michaelnebel — APPROVED** [review]
C# 👍

**Review by hvitved — APPROVED** [review]
C# + Ruby 👍

**Review by aschackmull — APPROVED** [review]
Java 👍

File: python/ql/lib/change-notes/2024-09-03-outdated-deprecations.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+category: breaking
+---
+* Deleted the deprecated `explorationLimit` predicate from `DataFlow::Configuration`, use `FlowExploration<explorationLimit>` instead.
+* Deleted the deprecated `semmle.python.RegexTreeView` module, use `semmle.python.regexp.RegexTreeView` instead.
+* Deleted the deprecated `RegexString` class from  `regex.qll`.
+* Deleted the deprecated `Regex` class, use `Regex` instead.
```
[the remark this answers is not in the document]
Comment (erik-krogh) [reply python/ql/lib/change-notes/2024-09-03-outdated-deprecations.md]: Yes. That was a mistake.

**Review by owen-mc — COMMENTED** [review]
Go again 👍🏻

**Review by tausbn — APPROVED** [review]

**Pull request merged**